utilities/os_utils.py

```
@@ -285,3 +286,24 @@ def _format_data_source_name(preference_name: str) -> str:
+    Returns:
+        dict: Latest supported OS dict (the os_values payload) or raises an exception.
+
+    Raises:
+        OsDictNotFoundError: If no os matched.
+    """
+    for matrix in os_matrix:
+        for os_values in matrix.values():
+            if os_values.get("latest_released"):
+                return os_values
+
+    raise OsDictNotFoundError(f"No OS is marked as 'latest_released': {os_matrix}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)